.github/scripts/security-guard.sh

```
@@ -0,0 +1,113 @@
+  read -r -d '' -a runner_args <<< "$runner"
+  log "running mandatory secret scan"
+  "${runner_args[@]}" secret scan pre-commit
+}
+
+run_fast_optional_checks() {
+  if [ "$RUN_FAST" -ne 1 ]; then
+    log "FAST checks disabled (SECURITY_GUARD_RUN_FAST=0)"
+    return 0
+  fi
+
+  local files cmd runner
```

Shellcheck (0.11.0)

[warning] cmd appears unused. Verify use (or export if used externally).

(SC2034)